java/statestore/src/main/java/sleeper/statestore/s3/S3FileReferenceStore.java

```diff
@@ -213,6 +214,8 @@ public void atomicallyUpdateFilesToReadyForGCAndCreateNewActiveFiles(
                     return "Files in filesToBeMarkedReadyForGC should be active: file " + filename + " is not active in partition " + partitionId;
                 } else if (!jobId.equals(activePartitionFiles.get(partitionId + DELIMITER + filename).getJobId())) {
                     return "Files in filesToBeMarkedReadyForGC should be assigned jobId " + jobId;
+                } else if (newFilenames.contains(filename)) {
+                    return "Files in filesToBeMarkedReadyForGC has same filename as new file: " + filename;
```

Comment:
I think this check means that the logic below for resolving update references is no longer needed, with the variable filenamesWithUpdatedReferences. We can remove the code for adding references to existing files.

Reply:
Thanks, removed the code for updating existing files